nearcore/src/config.rs

```diff
@@ -788,6 +827,17 @@ impl NearConfig {
     }
 }
 
+pub fn update_with(
+    store_config: near_store::StoreConfig,
+    config: &Config,
+) -> near_store::StoreConfig {
+    let mut store_config = store_config.clone();
+    store_config.enable_statistics = config.enable_rocksdb_statistics;
+    store_config.max_open_files = config.store.max_open_files;
+    store_config.col_state_cache_size = config.store.col_state_cache_size;
+    store_config
+}
```

Comment:
This would work better as `Into<near_store::StoreConfig>` implementation IMO. Then add `pub fn with_read_only(self, read_only: bool) -> Self` method to `near_store::StoreConfig` which changes the value of the field and you can do `near_store::StoreConfig::from(near_config.store).read_only(true)`. This would also work in `get_version` where you’d just do `store_config.clone().read_only(true)`.